specification/terraform/Microsoft.AzureTerraform.Management/routes.tsp

```diff
@@ -2,52 +2,67 @@ import "@azure-tools/typespec-azure-core";
-  }
-)
-op exportTerraform(
-  ...ApiVersionParameter,
-  ...SubscriptionIdParameter,
+@doc("Parameters common to all ExportTerraform operations")
+model CommonExportTerraformParameters {
+  ...ApiVersionParameter;
+  ...SubscriptionIdParameter;
+}
+
+#suppress "@azure-tools/typespec-azure-resource-manager/arm-resource-interface-requires-decorator" "Cannot use @armResourceOperations, the implied @autoRoute produces the wrong route for GET /operationStatuses/{operationId}"
+@route("/subscriptions/{subscriptionId}/providers/Microsoft.AzureTerraform")
+interface ExportTerraform {
+  // Cannot use Azure.ResourceManager.ArmProviderActionAsync due to error during .NET SDK code generation:
+  // Unknown identifier SubscriptionActionScope
+  #suppress "@azure-tools/typespec-azure-core/no-openapi" "Need to override @operationId to fix LintDiff"
+  @armResourceAction(ExportResult)
+  @doc("Exports the Terraform configuration of the specified resource(s).")
+  @route("/exportTerraform")
+  @post
+  @tag("ExportTerraform")
+  @operationId("ExportTerraform_Export")
+  exportTerraform(
+    ...ApiVersionParameter,
+    ...SubscriptionIdParameter,
 
-  @doc("The export parameter")
-  @body
-  exportParameter: BaseExportModel,
-): ArmAcceptedLroResponse<
-  "Export request accepted.",
-  ArmCombinedLroHeaders<ArmOperationStatus, OperationStatus> &
-    Azure.Core.Foundations.RetryAfterHeader
-> | ErrorResponse;
+    @doc("The export parameter")
+    @body
+    exportParameter: BaseExportModel,
+  ): ArmAcceptedLroResponse<
+    "Resource operation accepted.",
+    ArmLroLocationHeader<
+      Azure.Core.StatusMonitorPollingOptions<ArmOperationStatus>,
+      OperationStatus,
+      string
+    > &
+      Azure.Core.Foundations.RetryAfterHeader
+  > | ErrorResponse;
 
-#suppress "@azure-tools/typespec-azure-resource-manager/arm-resource-interface-requires-decorator" "Cannot use @armResourceOperations decorator here, the auto-generated routes do not match feature requirements"
-interface OperationStatuses {
-  #suppress "@azure-tools/typespec-azure-resource-manager/no-response-body" "Body of 202 is not empty: not compatible with API requirements"
+  #suppress "@azure-tools/typespec-azure-resource-manager/no-response-body" "202 response of this API has body"
+  #suppress "@azure-tools/typespec-azure-core/no-openapi" "Need to override @operationId to fix LintDiff"
   @doc("Get the status of a long running azure asynchronous operation.")
-  @route("/subscriptions/{subscriptionId}/providers/Microsoft.AzureTerraform/operationStatuses/{operationId}")
+  @route("/operationStatuses/{operationId}")
   @get
   @tag("OperationStatuses")
-  @armResourceRead(ArmResponse<OperationStatus>)
-  get(
-    ...OperationIdParameter,
-    ...ApiVersionParameter,
-    ...SubscriptionIdParameter,
-  ): OperationStatus | InProgressOperationStatus | ErrorResponse;
+  @operationId("ExportTerraform_GetOperationStatuses")
+  operationStatuses is ArmReadOperation<
```

Comment:
i think `getOperationStatus` is a better name for an operation like this, an operation name should always have a verb in it.
and the `operationId` would not be needed (unless there is a breaking change issue).
This operation is returning one `OperationStatus` instead of returning a list of statuses, therefore I think it should be a singular form.

Reply:
Good idea, renamed in e669bc3b30.